### ISSUE TITLE
Gas price must be set in the client options when auto gas is used

### ISSUE THREAD

**Issue by AdrienLemaire** [strangelove-ventures/wadesta#106]
TLDR: I cannot instantiate a contract using `useInstantiateContract` or `useSigningClients` because it keeps calling `cosmWasm.instantiate` with auto gas and no gas price somewhere in the back, disregarding my configuration and client options.

--- 

Related issue: #7 

I'm getting the following error regarding gas price when trying to instantiate a contract.

```console
ContractCreate.tsx:23 Error: Gas price must be set in the client options when auto gas is used.
    at assertDefined (assert.ts:10:11)
    at _SigningCosmWasmClient.signAndBroadcast (signingcosmwasmclient.ts:548:7)
    at _SigningCosmWasmClient.instantiate (signingcosmwasmclient.ts:321:31)
    at lt3 (index.mjs:1:28075)
    at Object.mutationFn (index.mjs:1:41853)
    at Object.fn (mutation.ts:179:31)
    at run3 (retryer.ts:147:31)
    at createRetryer (retryer.ts:204:5)
    at executeMutation (mutation.ts:174:22)
    at Mutation.execute (mutation.ts:216:26)
```

This is strange, since gas price was already configured in the chain:

```ts
export const customChains = defineChains({
  chainName: {
    gas: {
      price: "1000000",
      denom: "udenom",
    },
    ...
});
configureGraz({
  defaultChain: customChains.chainName,
});
```

Here's what I tried with useInstantiateContract:

```ts
export default function ContractCreate() {
  const { instantiateContract } = useInstantiateContract({
    codeId: 5,
    onSuccess: ({ contractAddress }) => console.log("Address: ", contractAddress),
  });

  const handleClick = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    const label = (event.currentTarget.elements.namedItem("label") as HTMLInputElement).value;
    const program = (event.currentTarget.elements.namedItem("program") as HTMLTextAreaElement).value;
    const msg = { program: btoa(program) };
    const options = { admin: ADMIN_ADDRESS };
    instantiateContract({ msg, options, label, });
  };

  return (
    <div>
      <h3>Create Contract</h3>
      <form onSubmit={handleClick}>
        <label htmlFor="program">program</label>
        <textarea id="program" />
        <label htmlFor="label">label</label>
        <input id="label" type="text" />
        <button type="submit">Create Contract</button>
      </form>
    </div>
  );
}
```

I then tried with cosmWasm from useSigningClients:

```ts
export default function ContractCreate() {
  const { signer } = useOfflineSigners();
  const signingClientsArgs = {
    cosmWasmSignerOptions: {
      gasPrice: GasPrice.fromString("0.012udenom"),
    },
    rpc: "https://api.somewhere.network:443/rpc",
    offlineSignerAuto: signer!,
  };
  const { data } = useSigningClients(signer ? signingClientsArgs : undefined);
  const { cosmWasm } = data || {};

  const handleClick = (event: React.FormEvent<HTMLFormElement>) => {
    if (!cosmWasm) return;
    event.preventDefault();
    const label = (event.currentTarget.elements.namedItem("label") as HTMLInputElement).value;
    const program = (event.currentTarget.elements.namedItem("program") as HTMLTextAreaElement).value;
    const msg = { program: btoa(program) };
    const options = { admin: ADMIN_ADDRESS };
    cosmWasm.instantiate(ADMIN_ADDRESS, 5, msg, label, 1000000, options);
  };

  return (
    <div>
      <h3>Create Contract</h3>
      <form onSubmit={handleClick}>
        <label htmlFor="program">program</label>
        <textarea id="program" />
        <label htmlFor="label">label</label>
        <input id="label" type="text" />
        <button type="submit">Create Contract</button>
      </form>
    </div>
  );
}
```

and getting the same error
```console
assert.ts:10 Uncaught (in promise) Error: Gas price must be set in the client options when auto gas is used.
    at assertDefined (assert.ts:10:11)
    at _SigningCosmWasmClient.signAndBroadcast (signingcosmwasmclient.ts:548:7)
    at _SigningCosmWasmClient.instantiate (signingcosmwasmclient.ts:321:31)
    at handleClick (LawStoneCreate.tsx:61:14)
    at HTMLUnknownElement.callCallback2 (react-dom.development.js:4164:14)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:4213:16)
    at invokeGuardedCallback (react-dom.development.js:4277:31)
    at invokeGuardedCallbackAndCatchFirstError (react-dom.development.js:4291:25)
    at executeDispatch (react-dom.development.js:9041:3)
    at processDispatchQueueItemsInOrder (react-dom.development.js:9073:7)
```

Note that I also called `cosmWasm.instantiate` with fee `1000000` instead of `auto`, yet still get the same error

**Comment by codingki** [comment]
This is fixed, in last version `0.0.49`
```ts
useCosmWasmSigningClient({
  opts: {
    gasPrice: GasPrice.fromString("0.012udenom"),
  }
})
```
